sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -218,6 +252,11 @@ object BinaryArithmetic {
   """)
 case class Add(left: Expression, right: Expression) extends BinaryArithmetic {
 
+  override def nullable: Boolean = dataType match {
+    case CalendarIntervalType if !checkOverflow => true
```

Comment:
shall we make the overflow behavior consistent? e.g. other numeric types follow the java overflow behavior and interval returns null for overflow, which is inconsistent.

Reply:
Yes, the current behavior of master is separating a) decimal(which is null for overflow) from b) other types. This pr (so far)is just adding intervals to group a). We may reach an agreement on which way to follow first.

Reply:
in spark 2.4, do we have internal arithmetic operations? The non-ANSI behavior should follow the old behavior.

Reply:
Yes, we have.
we now have `+/-/unaray_-` in java overflow behavior(2.4 or maybe earlier)
and we have '*/-' in null for overflow behavior (3.0)

Reply:
then let's be consistent and follow java overflow behavior when ansi is false.

Reply:
```
spark-sql> select cast('128' as tinyint);
NULL
Time taken: 0.029 seconds, Fetched 1 row(s)
spark-sql> select cast(128 as tinyint);
-128
```
not quite related to this pr, the cast logic seems unconsisent too

Reply:
cast from string and cast from int are different and don't need to be consistent. But interval +- and interval */ should be consistent.

Reply:
changed behavior to a) ANSI on: exception,  b) ANSI off: java style overflow